src/fuzzfetch/fetch.py

```diff
@@ -349,9 +358,51 @@ def __init__(self, target, branch, build, flags, platform=None):
             self._task = build
 
         else:
-            self._task = BuildTask(build, branch, self._flags, self._platform)
-
             now = datetime.now(timezone('UTC'))
+            try:
+                self._task = BuildTask(build, branch, self._flags, self._platform)
+            except FetcherException:
+                if not nearest:
+                    raise
+                else:
+                    asc = nearest == Fetcher.BUILD_ORDER_ASC
+                    start = None
+                    if build == 'latest':
+                        start = now
+                    elif re.match(r'^[0-9[a-f]{12,40}$', build):
+                        start = _get_rev_date(build)
+
+                    elif re.match(r'^[0-9]{4}-[0-9]{2}-[0-9]{2}$', build):
```

Comment:
(Here and elsewhere) to check if a match was made, always use `re.match() is None`. Checking for truthiness does some extra work, and will be wrong for an empty match (even though that doesn't apply here, it's good practice).

Reply:
Done